application/common/helpers/Utils.php

```diff
@@ -278,4 +283,73 @@ public static function getRandomDigits($length = 4)
         return $randomDigits;
     }
 
+    /**
+     * Call reCaptcha API to verify response token
+     * @param string $verificationToken
+     * @param string $ipAddress
+     * @return bool
+     * @throws \Exception
+     */
+    public static function isRecaptchaResponseValid($verificationToken, $ipAddress)
+    {
+        $recaptcha = new \ReCaptcha\ReCaptcha(\Yii::$app->params['recaptcha']['secretKey']);
+        $response = $recaptcha->verify($verificationToken, $ipAddress);
+
+        if ($response->isSuccess()) {
+            return true;
+        }
+
+        \Yii::error([
+            'action' => __METHOD__,
+            'status' => 'error',
+            'error' => Json::encode($response->getErrorCodes()),
+        ]);
+        throw new BadRequestHttpException('Unable to verify recaptcha', 1462904023);
+    }
+
+    /**
+     * Get Client IP address by looking through headers for proxied requests
+     * @param Request $request
+     * @return string
+     */
+    public static function getClientIp(Request $request)
+    {
+        $checkHeaders = [
+            'X-Forwarded-For',
+            'X-Forwarded',
+            'X-Cluster-Client-Ip',
+            'Client-Ip',
+        ];
+
+        $ipAddress = $request->userIP;
+
+        $requestHeaders = $request->getHeaders();
+        foreach ($checkHeaders as $header) {
+            if ($requestHeaders->has($header)) {
+                $ip = trim(current(explode(',', $requestHeaders->get($header))));
+                if (self::isValidIpAddress($ip)) {
+                    $ipAddress = $ip;
+                    break;
+                }
+            }
+        }
+
+        return $ipAddress;
+    }
+
+    /**
+     * Check that a given string is a valid IP address
+     *
+     * @param  string  $ip
+     * @return boolean
+     */
+    public static function isValidIpAddress($ip)
+    {
+        $flags = FILTER_FLAG_IPV4 | FILTER_FLAG_IPV6;
+        if (filter_var($ip, FILTER_VALIDATE_IP, $flags) === false) {
+            return false;
+        }
+        return true;
```

Comment:
If you wanted, you could reduce these few lines to...
`return (filter_var($ip, FILTER_VALIDATE_IP, $flags) !== false);`

I'm not sure if that's more or less readable, though, which seems more important than conciseness.